src/coreclr/jit/emit.cpp

```diff
@@ -6151,6 +6158,11 @@ unsigned emitter::emitEndCodeGen(Compiler* comp,
     // Assign the real prolog size
     *prologSize = emitCodeOffset(emitPrologIG, emitPrologEndPos);
 
+    // Ensure that any attempt to write code after this point will fail
+    writeableOffset = 0;
+    // Notify the EE that all code was written. It can switch off writeable code memory.
+    emitCmpHandle->doneWritingCode();
```

Comment:
This isn't guaranteed to be called, if the jit decides to abandon this jit attempt (either because of minopts fallback or perhaps inability to use RIP addressing or whatnot).

Is this a todo?

We should either use the `runWithErrorTrap` wrapper here (which will require refactoring the jit code), or else have the runtime side keep track of whether the jit ever called `doneWritingCode` and do suitable cleanup when it has the chance, if the jit didn't do it.

Reply:
Ah, I knew I've forgotten something. Of course it should be inside the runWithErrorTrap.